config-reloader/template/template.go

```diff
@@ -0,0 +1,144 @@
+package template
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	konfig "sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/yaml"
+)
+
+var k8sClient client.Reader
+
+// Render is a go template rendering function it includes all the sprig lib functions
+// as well as some extras like a k8sLookup function to get values from k8s objects
+// you can access environment variables from the template under .Env
+// The passed values will be available under .Values in the templates
+func Render(out io.Writer, tmpl string, values interface{}) error {
+	t := template.Must(New("tmpl").Parse(tmpl))
+	return t.Execute(out, map[string]interface{}{
+		"Env":    envMap(),
+		"Values": values,
+	})
+}
+
+func Must(t *template.Template, err error) *template.Template {
```

Comment:
Instead of using panic, consider returning the error for the caller to handle.

Reply:
This is to be a drop in replacement of the `Must` function in the stdlib template package to avoid mutch refactoring, we can change it later but I'd rather keep it like this in this PR.